cpp/src/rolling/rolling.cu

```diff
@@ -38,14 +39,116 @@ namespace experimental {
 namespace detail {
 
 namespace { // anonymous
+// Meant for only count operation
+template <typename InputType, typename OutputType, typename agg_op, aggregation::Kind op, bool has_nulls>
+std::enable_if_t<op == aggregation::COUNT, bool>
+__device__
+specific_rolling_kernel(column_device_view input,
+                        mutable_column_device_view output,
+                        size_type start_index,
+                        size_type end_index,
+                        size_type current_index,
+                        size_type min_periods,
+                        InputType identity) {
+
+    // declare this as volatile to avoid some compiler optimizations that lead to incorrect results
+    // for CUDA 10.0 and below (fixed in CUDA 10.1)
+    volatile cudf::size_type count = 0;
+    
+    for (size_type j = start_index; j < end_index; j++) {
+        if (!has_nulls || input.is_valid(j)) {
+            count++;
+        }
+    }
+   
+    bool output_is_valid = (count >= min_periods);
+    output.element<OutputType>(current_index) = count;
+
+    return output_is_valid;
+}
+
+// Meant for only ArgMin and ArgMax for string_view
+template <typename InputType, typename OutputType, typename agg_op, aggregation::Kind op, bool has_nulls>
```

Comment:
I'm curious why this is specialized for strings.  Is there something about the agg_op interface for min/max that doesn't work for all types?

Reply:
As of now we support MIN, MAX, COUNT for String and ArgMin and ArgMax is used to collect those indices and gather. And we don't support ArgMin and ArgMax as such.